lib/ansible/plugins/terminal/ironware.py

```diff
@@ -30,7 +30,7 @@
 class TerminalModule(TerminalBase):
 
     terminal_stdout_re = [
-        re.compile(r"[\r\n]?(?:\w+@)?[\w+\-\.:\/\[\]]+(?:\([^\)]+\)){,3}(?:>|#) ?$")
+        re.compile(r"[\r\n](?:\w+@)?[\w+\-\.:\/\[\]]+(?:\([^\)]+\)){,3}(?:>|#) ?$")
```

Comment:
Actually this doesn't seem to work for ironware, not sure exactly why not though. 

The full traceback is:
Traceback (most recent call last):
  File "/Users/paul.baker/ansible/bin/ansible-connection", line 70, in start
    self.connection._connect()
  File "/Users/paul.baker/ansible/lib/ansible/plugins/connection/network_cli.py", line 315, in _connect
    self.receive()
  File "/Users/paul.baker/ansible/lib/ansible/plugins/connection/network_cli.py", line 387, in receive
    data = self._ssh_shell.recv(256)
  File "/usr/local/lib/python2.7/site-packages/paramiko/channel.py", line 685, in recv
    raise socket.timeout()

Could it be that there is no newline character when we first open a connection?

When I SSH to the router this is what I see, and this does seem to match the RE above.
$ ssh paulb@rtr02.mel01
Password:
SSH@rtr02.mel01#

Reply:
@paulquack Did you happen to dig into this anymore, it definitely looks like that regex would match.  Are there any other non-printing characters before the prompt?  A bell `^G` or anything?